docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -400,6 +400,24 @@ specified in the following form:
     {"key": "header2", "value": "value2"}
   ]
 
+Envoy supports adding static and dynamic values to the request headers. The dynamic values supported are:
+
+%CLIENT_IP%
+   The Original Client IP which is already added by envoy as an  
+   :ref:`X-Forwarded-For <config_http_conn_man_headers_x-forwarded-for>` request header. 
+
+%PROTOCOL%
+    The Original Protocol which is already added by envoy as an 
+    :ref:`X-Forwarded-Proto <config_http_conn_man_headers_x-forwarded-proto>` request header. 
+
+An example for adding dynamic value to the request headers would be as follows:
+
+.. code-block:: json
+
+  [
+   {"key": "X-Client-IP", "value":"%CLIENT_IP%"}
```

Comment:
can you also add integration test to test this? (need to change integration test envoy config)